pyfarm/master/api/jobgroups.py

```diff
@@ -0,0 +1,355 @@
+
+                {
+                    "title": "Test Group",
+                    "user": "testuser",
+                    "main_jobtype": "Test JobType"
+                }
+
+
+            **Response**
+
+            .. sourcecode:: http
+
+                HTTP/1.1 201 CREATED
+                Content-Type: application/json
+
+                {
+                    "id": 2,
+                    "jobs": [],
+                    "user": "testuser",
+                    "main_jobtype": "Test JobType",
+                    "title": "Test Group"
+                }
+
+        :statuscode 201: a new job group was created
+        :statuscode 400: there was something wrong with the request (such as
+                         invalid columns being included)
+        """
+        username = g.json.pop("user")
+        user = User.query.filter_by(username=username).first()
+        if not user and AUTOCREATE_USERS:
+            user = User(username=username)
+            if AUTO_USERS_DEFAULT_DOMAIN:
+                user.email = username + "@" + AUTO_USERS_DEFAULT_DOMAIN
+            db.session.add(user)
+            logger.warning("User %s was autocreated on job group create",
+                           username)
+        elif not user:
+            return (jsonify(
+                error="User %s not found" % username), NOT_FOUND)
+
+        jobtype_name = g.json.pop("main_jobtype")
+        jobtype = JobType.query.filter_by(name=jobtype_name).first()
+        if not jobtype:
+            return (jsonify(
+                error="Jobtype %s not found" % jobtype_name), NOT_FOUND)
+
+        jobgroup = JobGroup(user=user, main_jobtype=jobtype, **g.json)
+        db.session.add(jobgroup)
+        db.session.commit()
+
+        jobgroup_data = jobgroup.to_dict()
+        del jobgroup_data["user_id"]
+        del jobgroup_data["main_jobtype_id"]
+        logger.info("Created job group %s: %r", jobgroup.title, jobgroup_data)
+
+        return jsonify(jobgroup_data), CREATED
+
+    def get(self):
+        """
+        A ``GET`` to this endpoint will return a list of known job groups.
+
+        .. http:get:: /api/v1/jobgroups/ HTTP/1.1
+
+            **Request**
+
+            .. sourcecode:: http
+
+                GET /api/v1/jobgroups/ HTTP/1.1
+                Accept: application/json
+
+            **Response**
+
+            .. sourcecode:: http
+
+                HTTP/1.1 200 OK
+                Content-Type: application/json
+
+                [
+                    {
+                        "id": 2,
+                        "user": "testuser",
+                        "main_jobtype": "Test JobType",
+                        "title": "Test Group"
+                    }
+                ]
+
+        :statuscode 200: no error
+        """
+        out = []
+        for jobgroup in JobGroup.query:
+            jobgroup_data = jobgroup.to_dict(
+                unpack_relationships=["user", "main_jobtype"])
+            del jobgroup_data["user_id"]
```

Comment:
Same function but won't cause the whole request to fail if `jobgroup_data` does not contain the keys we're trying to remove:

```python
jobgroup_data.pop("user_id', None)
jobgroup_data.pop("main_jobtype_id", None)
```